ethToElrond/v2/ethToElrondBridgeExecutor_test.go

```diff
@@ -0,0 +1,601 @@
+		assert.Equal(t, errNilEthereumClient, err)
+	})
+	t.Run("nil topology provider should error", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockEthToElrondExecutorArgs()
+		args.TopologyProvider = nil
+		executor, err := NewEthToElrondBridgeExecutor(args)
+
+		assert.True(t, check.IfNil(executor))
+		assert.Equal(t, errNilTopologyProvider, err)
+	})
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockEthToElrondExecutorArgs()
+		executor, err := NewEthToElrondBridgeExecutor(args)
+
+		assert.False(t, check.IfNil(executor))
+		assert.Nil(t, err)
+	})
+}
+
+func TestEthToElrondBridgeExecutor_GetLogger(t *testing.T) {
+	t.Parallel()
+
+	args := createMockEthToElrondExecutorArgs()
+	executor, _ := NewEthToElrondBridgeExecutor(args)
+
+	assert.True(t, args.Log == executor.GetLogger()) //pointer testing
+}
+
+func TestEthToElrondBridgeExecutor_MyTurnAsLeader(t *testing.T) {
+	t.Parallel()
+
+	args := createMockEthToElrondExecutorArgs()
+	wasCalled := false
+	args.TopologyProvider = &bridgeV2.TopologyProviderStub{
+		MyTurnAsLeaderCalled: func() bool {
+			wasCalled = true
+			return true
+		},
+	}
+
+	executor, _ := NewEthToElrondBridgeExecutor(args)
+	assert.True(t, executor.MyTurnAsLeader())
+	assert.True(t, wasCalled)
+}
+
+func TestEthToElrondBridgeExecutor_GetAndStoreActionID(t *testing.T) {
+	t.Parallel()
+
+	t.Run("nil batch should error", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockEthToElrondExecutorArgs()
+		executor, _ := NewEthToElrondBridgeExecutor(args)
+
+		actionID, err := executor.GetAndStoreActionID(context.Background())
+		assert.Zero(t, actionID)
+		assert.Equal(t, errNilBatch, err)
+	})
+	t.Run("elrond client errors", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockEthToElrondExecutorArgs()
+		expectedErr := errors.New("expected error")
+		providedBatch := &clients.TransferBatch{}
+
+		args.ElrondClient = &bridgeV2.ElrondClientStub{
+			GetActionIDForProposeTransferCalled: func(ctx context.Context, batch *clients.TransferBatch) (uint64, error) {
+				assert.True(t, providedBatch == batch)
+				return 0, expectedErr
+			},
+		}
+		executor, _ := NewEthToElrondBridgeExecutor(args)
+		executor.batch = providedBatch
+
+		actionID, err := executor.GetAndStoreActionID(context.Background())
+		assert.Zero(t, actionID)
+		assert.Equal(t, expectedErr, err)
+	})
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockEthToElrondExecutorArgs()
+		providedBatch := &clients.TransferBatch{}
+		providedActionID := uint64(48939)
+
+		args.ElrondClient = &bridgeV2.ElrondClientStub{
+			GetActionIDForProposeTransferCalled: func(ctx context.Context, batch *clients.TransferBatch) (uint64, error) {
+				assert.True(t, providedBatch == batch)
+				return providedActionID, nil
+			},
+		}
+		executor, _ := NewEthToElrondBridgeExecutor(args)
+		executor.batch = providedBatch
+
+		actionID, err := executor.GetAndStoreActionID(context.Background())
```

Comment:
Should we also add a check for the actual **store** part, not only for the getter?
E.g.:
```go
		assert.NotEqual(t, providedActionID, executor.actionID)
		actionID, err := executor.GetAndStoreActionID(context.Background())
		assert.Equal(t, providedActionID, executor.actionID)
```

Reply:
done